README.md

```diff
@@ -1,2 +1,233 @@
-# messaging
-Messaging for Web3
+# Anonymous Communication Protocol
+
+Messaging for Web3.
+
+## Motivation
+In the current decentralised application landscape, we are seeing projects
+struggle to achieve mainstream adoption. We believe that the underlying
+protocols do not have sufficient capability to enable the necessary level of
+adoption. Part of the problem is that these applications often require the
+exchange of transient messages; however, it does not make sense to transmit
+these messages via a blockchain. Gavin Wood realised this problem in the early
+days of Ethereum, and suggested that DApps would require a decentralised
+messaging protocol that would provide this capability. This protocol is called
+Whisper.
+
+Unfortunately, the evolution and adoption of Whisper has been stunted, which is
+despite the rapid advancement of applications. The lack of development means
+that Whisper is not able to provide the scaling that we need, nor is it feasible
+for projects to create their own bespoke messaging protocol. What we really need
+is a new protocol to handle transient messaging at scale. We believe this is a
+necessary cornerstone for enabling the mainstream adoption of DApps.
+
+## Project
+We would like to gather a number of projects together to align and support this protocol:
+- Researchers
+   - Academia
+- Web 3 projects
+   - Protocol implementers
+- Core dev teams from Web 3 space
+- Application builders
+  - User messaging application
+  - State channels
+  - Streaming protocols
+  - Other applications requiring transient messaging
+
+The idea is to start with gathering requirements and aligning on goals, the Web3
+Foundation (W3F) has discussed with a number of projects to better understand
+what the needs are. W3F has also started to do an initial exploration of
+potential components to be used in the protocol.
+
+After the initial round of asynchronous information gathering W3F aims to
+organise a workshop for all potential initial contributors.
+
+### Project contributors
+- Web3 Foundation
+- Status
+- Validity Labs
+
+## Protocol requirements
+
+The purpose of an anonymous communication (AC) protocol is to ensure metadata
+isn't leaked when messages are communicated between peers. It deals with how
+messages are transported, and not what is in them.
+
+An overarching principle is to keep things as simple as possible, and try to
+factor out subproblems to separate layers as much as possible. When in doubt,
+leave it out.
+
+Metadata that we want to protect are:
+
+**1. Sender Anonymity** (who sent a message?)
+**2. Receiver Anonymity** (who read a message?)
+**3. Sender-Receiver Unlinkability** (who is talking to whom?)
+
+Primary consideration in threat model is:
+
+- Global Passive Adversary resistant (GPA, insight into whole network)
+
+Some consideration should also be given to a global active attacker (GAA), in
```

Comment:
Threat model should be more fleshed out, as well as a section on what class of attacks it should / will not / may protect against.